### PR TITLE
S2 profilepage read

### PR DESCRIPTION
**Pull request by DariaBezuhla**
* added READ function to the profile page
* improved logic for READ apps on user profile 
* added edit/delete button to the apps cards
* added counts of "your apps"

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -1,9 +1,39 @@
...
+        req.data = apps;
+      }
+      catch(error) {
+        console.log(error);
+        respondNoResourceFound(req, res);
+      }
+      next();
+    })
+  },
+
```
Comment (kimc0de) [inline controllers/userController.js:19]:
```suggestion
```

File: controllers/userController.js [hunk 2]
```diff
@@ -1,9 +1,39 @@
...
+  renderProfile: (req, res, next) => {
+    let userId = req.params.id;
+    
+    User.findById(userId)
+      .then(user => {
+        res.render("user/profile", {
+          user:user,
+          data: req.data,
+          app: req.app,
+
```
Comment (kimc0de) [inline controllers/userController.js:30]:
```suggestion
```

File: public/stylesheets/profile.css [hunk 3]
```diff
@@ -118,4 +130,11 @@
     margin-right: 3vh;
   }
```
Comment (kimc0de) [inline public/stylesheets/profile.css:132]:
```suggestion
```

File: public/stylesheets/profile.css [hunk 4]
```diff
@@ -118,4 +130,11 @@
...
 
+}
+
+@media (max-width: 768px) {
+  #profile-page .app-card-buttons {
+    margin-left: 1.8vh;
+  }
+
```
Comment (kimc0de) [inline public/stylesheets/profile.css:139]:
```suggestion
```

File: views/user/profile.ejs [hunk 5]
```diff
@@ -3,12 +3,13 @@
 
 <div id="profile-page" class="container d-flex flex-column align-items-center">
   
-  <img src="/images/greenpick/profilePic1.png">
-  <h2 id="user-name">Zekiye</h2>
+  <img id="profile-image" src="/images/greenpick/profilePic1.png">
+  <h2 id="user-name"><%= user.username %></h2>
+
```
Comment (kimc0de) [inline views/user/profile.ejs:8]:
```suggestion
```

**Comment by kimc0de** [comment]
@DariaBezuhla the responsive style of the card is currently still messed up.. could u fix it before merge?

File: views/user/profile.ejs [hunk 6]
```diff
@@ -3,12 +3,13 @@
```
Comment (kimc0de) [inline views/user/profile.ejs:3]:
```suggestion
```

File: views/user/profile.ejs [hunk 7]
```diff
@@ -22,12 +23,35 @@
...
     <div id="your-apps" class="app-list active">
       <div class="d-flex flex-row">
-        <p class="number-of-apps">0 </p>
+        <p class="number-of-apps"><%= data.length %></p>
         <p class="text-in-app-list">Wonderful! Those are sustainable apps that you have added.</p>
       </div>
+    
+      <div class="row" id="app-card">
+        <% data.forEach(app => { %>
+        <div class="col col-lg-4 col-md-6 col-12" onclick="location.href='/app/<%= app._id %>';">
```
Comment (kimc0de) [inline views/user/profile.ejs:35]:
```suggestion
        <div class="col col-lg-6 col-md-6 col-sm-12" onclick="location.href='/app/<%= app._id %>';">
```

Comment (kimc0de) [reply views/user/profile.ejs:35]: this will fix the responsive i think

File: controllers/userController.js (hunk 1 above)
Comment (kimc0de) [reply controllers/userController.js:19]: these suggestions are just extra empty lines that we can delete

File: views/user/profile.ejs (hunk 7 above)
Comment (DariaBezuhla) [reply views/user/profile.ejs:34]: oh no, the problem was different but i tried to change those first, and forgot to change them back as they were. thanks! :)

**Comment by DariaBezuhla** [comment]
have checked as well the cards, they are okay now

**Pull request merged**